### ISSUE TITLE
problem mounting with `el` option after unmounted

### ISSUE THREAD

**Issue by karladler** [single-spa/single-spa-vue#33]
When I add the `el` option the first time the instance is correctly rendered into the selected element.
But since vue replaces this element there is a problem mounting it again after unmounted, because the desired element does not exists anymore on the DOM.

> The provided element merely serves as a mounting point. Unlike in Vue 1.x, the mounted element will be replaced with Vue-generated DOM in all cases. It is therefore not recommended to mount the root instance to <html> or <body>.

In my case I just want to layout the positions of the single applications.

```html
  <div id="all-apps">
      <div id="navbar"></div>
      <div id="first" style="width: 50%;">
        <div></div>
      </div>
      <div id="second" style="width: 50%;">
        <div></div>
      </div>
    </div>
```

```javascript
const vueLifecycles = singleSpaVue({
  Vue,
  appOptions: {
    render: (h) => h(App),
    router,
    el: '#first > div'
  },
});
```

**Comment by joeldenning** [comment]
The `<div class="single-spa-container">` shown below is how this avoided when the `el` option is not provided:

https://github.com/single-spa/single-spa-vue/blob/bd25b1f208d0c059a39a6a2d91f1c4524917f65e/src/single-spa-vue.js#L64-L

We could perhaps try to do something similar when `#el` is provided. Do you have any ideas about how to solve this?

**Comment by jualoppaz** [comment]
Hi @karladler ! As you want to make a custom layout I think this is vue matter and not single-spa's.

I had to do a custom configuration in my root application and my  single-spa-vue app for keep the anchor all time for un/mount so many times I want.

You can see this strategy in next files:

https://github.com/jualoppaz/single-spa-vue-app/blob/0723ce4b073c3e31b94bb39f134d8128f5d10902/src/singleSpaEntry.js#L21-L28

https://github.com/jualoppaz/single-spa-login-example-with-npm-packages/blob/e73d0dd972726bcbfafe2700c5a62700cd3bf078/index.html#L16

**Comment by madeofsun** [comment]
> The `<div class="single-spa-container">` shown below is how this avoided when the `el` option is not provided:
> 
> https://github.com/single-spa/single-spa-vue/blob/bd25b1f208d0c059a39a6a2d91f1c4524917f65e/src/single-spa-vue.js#L64-L71
> 
> We could perhaps try to do something similar when `#el` is provided. Do you have any ideas about how to solve this?

Spent quite a lot time trying to understand why error happens on unmount stage, when parcel is mounted by `single-spa-react/parcel`. I am not common with Vue and did not know that it replaces the element instead of appending, which causes React to crash.

Ended up with similar solution:
```
const vueLifecycles = singleSpaVue({
  Vue,
  appOptions: {
    render: (h) => h(App)
  }
});

export const bootstrap = vueLifecycles.bootstrap;
export const mount = (props) => {
  if (props.domElement && props.append) {
    const el = document.createElement('div');
    props.domElement.appendChild(el);
    props.domElement = el;
  }
  return vueLifecycles.mount(props);
};
export const unmount = vueLifecycles.unmount;
```

**Comment by joeldenning** [comment]
I'm very open to figuring out a better way of handling these situations - perhaps automatically appending the wrapper container when the `#el` option is provided.

**Comment by maurer2** [comment]
Hello, we worked around the issue by creating a new wrapper-div with a random id during the mount phase, which gets appended to our main-content. This elements gets replaced with the single-spa app by Vue. Unfortunately, we can't access the new element for removal after it was replaced. So as a workaround, we empty the wrapper-div manually on unmount. The same logic is then applied to the new app.

HTML
```
<body class="wrapper">
    <div class="sidebar-wrapper">
        <div id="sidebar"></div>
    </div>
    <main class="main-wrapper" id="main-wrapper"></main>
</body>
``` 

TS
```
...
import nanoid from 'nanoid'

Vue.config.productionTip = false
Vue.config.devtools = true

const createComponentRootElement = (rootId: string): void => {
  const spaWrapper: HTMLElement = document.getElementById(rootId) || document.body

  const newComponent: HTMLElement = Object.assign(
    document.createElement('div'), {
      id: componentId
    }
  )
  spaWrapper.appendChild(newComponent)
}

const clearComponentRootElement = (rootId: string): void => {
  const spaWrapper: HTMLElement = document.getElementById(rootId) || document.body

  while (spaWrapper.firstChild) {
    spaWrapper.removeChild(spaWrapper.firstChild)
  }
}

const rootClass: string = 'main-wrapper'
const componentId: string = 'id' + nanoid().toLocaleLowerCase()

const vueLifecycles = singleSpaVue({
  Vue,
  appOptions: {
    render: (h:any) => h(Stub),
    router,
    el: `#${componentId}`
  }
})

export const bootstrap = [
  vueLifecycles.bootstrap
]

export const mount = [
  () => {
    createComponentRootElement(rootClass)

    return Promise.resolve()
  },
  vueLifecycles.mount
]

export const unmount = [
  vueLifecycles.unmount,
  () => {
    clearComponentRootElement(rootClass)

    return Promise.resolve()
  }
]

```

**Comment by joeldenning** [comment]
I just submitted a fix in #34

**Comment by joeldenning** [comment]
This is now supported in https://github.com/single-spa/single-spa-vue/releases/tag/v1.8.0
